### PR TITLE
control-service: add OAuth2 enable/disable flag

### PR DESCRIPTION
**Pull request by mrMoZ1**
why: As a part of https://github.com/vmware/versatile-data-kit/pull/755 we introduced a new authentication method.
This change adds another flag which disables/enables OAuth2 authentication
as requested here: https://github.com/vmware/versatile-data-kit/pull/755#discussion_r823978940
This pull request should be reviewed after https://github.com/vmware/versatile-data-kit/pull/755 is.

what: Added a new property and logic that adds OAuth2 to the control-service security filters if enabled.
OAuth2 is enabled by default - this setting shouldn't introduce any regressions to other existing code.
Security is disabled by default still.

testing: On a locally running control-service made sure that endpoints are secure
and proper filters where invoked when calling with and without an authenticated OAuth2 client through vdkcli:
`vdk execute --list -t supercollider -n job -u http://localhost:8092`

### CODE REVIEW

**Review by doks5 — APPROVED** [review]
LGTM

File: projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties [hunk 1]
```diff
@@ -149,3 +149,6 @@ datajobs.security.kerberos.kerberosPrincipal=""
 datajobs.security.kerberos.keytabFileLocation=
 datajobs.security.kerberos.krb5ConfigLocation=
 datajobs.security.kerberos.enabled=false
+
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:152]:
```#Settings for Kerberos authorization provider.
datajobs.security.kerberos.kerberosPrincipal=""
```
Principal, and the rest of the properties you can trace, are actually used to construct KerberosServiceAuthenticationProvider (SunJaasKerberosTicketValidator);
Those properties have to do with authentication rather authorization.

Authorization is performed via configurable authorization webhook, for example find configurables:
```
featureflag.authorization.enabled=true
datajobs.authorization.webhook.endpoint=https://httpbin.org/post
```

**Review by ivakoleva — CHANGES_REQUESTED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties [hunk 2]
```diff
@@ -149,3 +149,6 @@ datajobs.security.kerberos.kerberosPrincipal=""
...
+#Settings for OAuth2 authorization provider
+datajobs.security.oauth2.enabled=true
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:154]: is this a new property? don't we have already configurable oauth2? could you link to existing oauth2 configurations?

Comment (mrMoZ1) [reply projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:154]: @tozka requested to have 2 separate flags for kerberos and oauth2. This commit introduces the flag specifically for oAuth2.

Comment (ivakoleva) [reply projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:154]: I see, is it oauth2 authentication?
Could you link to existing oauth2 configurations?

Comment (mrMoZ1) [reply projects/control-service/projects/pipelines_control_service/src/main/resources/application.properties:154]: https://github.com/vmware/versatile-data-kit/blob/830e92432313a7739559659424cc24fd81a0f868/projects/control-service/projects/pipelines_control_service/src/main/resources/application-dev.properties#L38-L40

**Review by tozka — APPROVED** [review]
I think you need tests. 
 - what if security.enabled is false 
 - what if security.enabled is true and oath2 and kerberos are false
 - what if security.enabled is true and oath2 is true and kerberos are false
 - what if security.enabled is true and oath2 is false and kerberos are true

I am a bit suspicious the configuration will work if all cases correctly. 

I am going to give you approval, so if you plan to cover them in integration tests

File: projects/control-service/projects/base/src/main/java/com/vmware/taurus/base/FeatureFlags.java [hunk 3]
```diff
@@ -29,4 +29,7 @@
    @Value("${datajobs.security.kerberos.enabled:false}")
    boolean krbAuthEnabled = false;
 
+   @Value("${datajobs.security.oauth2.enabled}")
+   boolean oAuth2Enabled = true;
```
[the remark this answers is not in the document]
Comment (mrMoZ1) [reply projects/control-service/projects/base/src/main/java/com/vmware/taurus/base/FeatureFlags.java:33]: testing: On a locally running control-service made sure that endpoints are secure
and proper filters where invoked when calling with and without an authenticated OAuth2 client through vdkcli:
vdk execute --list -t supercollider -n job -u http://localhost:8092

[the remark this answers is not in the document]
Comment (ivakoleva) [reply projects/control-service/projects/base/src/main/java/com/vmware/taurus/base/FeatureFlags.java:34]: added default value, matching the securityEnabled one

File: projects/control-service/projects/base/src/main/java/com/vmware/taurus/security/SecurityConfiguration.java [hunk 4]
```diff
@@ -138,9 +138,11 @@ private void enableSecurity(HttpSecurity http) throws Exception {
                             .hasAnyAuthority(authorizedRoles.toArray(String[]::new));
                   }
                   authorizeRequests.anyRequest().authenticated();
-              })
-              .oauth2ResourceServer().jwt()
-              .jwtAuthenticationConverter(jwtAuthenticationConverter());
+              });
+
+        if (featureFlags.isOAuth2Enabled()) {
```
[the remark this answers is not in the document]
Comment (ivakoleva) [reply projects/control-service/projects/base/src/main/java/com/vmware/taurus/security/SecurityConfiguration.java:143]: will be covered in an additional PR;

**Review by ivakoleva — APPROVED** [review]

**Pull request merged**